src/julia-syntax.scm

```diff
@@ -1615,7 +1615,7 @@
                            (_do_while
 			    (block ,body
 				   (= ,next (call (top iterate) ,coll ,state)))
-			    (call (|.| (core Intrinsics) 'not_int) (call (core ===) ,next (null))))))))))))
+			    (call (top not_int) (call (core ===) ,next (null))))))))))))
```

Comment:
it's not a `top` function

Reply:
Yes, but it's visible from there and this leads to more compact IR. We do this in a couple other places. Would be good to eventually make this more accurate though.

Reply:
More compact in what way? Is there a measurable difference? It should resolve to the same binding (GlobalRef) pretty quickly. If that's not working for simple cases like this, we should probably fix that, since it's probably affecting other code too in significant ways.

> We do this in a couple other places. Would be good to eventually make this more accurate though.

AFAICT, there's one other place. Why don't we start now, with the known bugs, rather than intentionally introducing new ones hoping to "eventually" work on them?